templates/master/00-master/_base/files/usr-local-bin-openshift-recovery-tools-sh.yaml

```diff
@@ -20,15 +20,13 @@ contents:
 
     # download and test etcdctl from upstream release assets
     dl_etcdctl() {
-      GOOGLE_URL=https://storage.googleapis.com/etcd
-      DOWNLOAD_URL=${GOOGLE_URL}
-
-      echo "Downloading etcdctl binary.."
-      curl -s -L ${DOWNLOAD_URL}/${ETCD_VERSION}/etcd-${ETCD_VERSION}-linux-amd64.tar.gz -o $ASSET_DIR/tmp/etcd-${ETCD_VERSION}-linux-amd64.tar.gz \
-        && tar -xzf $ASSET_DIR/tmp/etcd-${ETCD_VERSION}-linux-amd64.tar.gz -C $ASSET_DIR/shared --strip-components=1 \
-        && mv $ASSET_DIR/shared/etcdctl $ASSET_DIR/bin \
-        && rm $ASSET_DIR/shared/etcd \
-        && $ASSET_DIR/bin/etcdctl version
+      local etcdimg="{{.Images.etcdKey}}"
```

Comment:
I can’t find where `.Images.etcdKey` comes from. Should it be `.Images.etcd` like here? https://github.com/openshift/machine-config-operator/blob/a55f2a6db546a801edd42b93e00b898b4f5588a9/pkg/operator/images.go#L30

Reply:
this is the template constant : https://github.com/openshift/machine-config-operator/blob/bd846958bc95d049547164046a962054fca093df/pkg/controller/template/constants.go#L5 which is gets rendered here: https://github.com/openshift/machine-config-operator/blob/61335d7845859313ff984f5a5b9d460ccccdd3e3/pkg/operator/bootstrap.go#L146